component/azstorage/block_blob.go

```diff
@@ -575,6 +577,31 @@ func (bb *BlockBlob) ReadToFile(name string, offset int64, count int64, fi *os.F
 		}
 	}
 
+	if bb.Config.validateMD5 {
+		// Compute md5 of local file
+		fileMD5, err := getMD5(fi)
+		if err != nil {
+			log.Warn("BlockBlob::ReadToFile : Failed to generate MD5 Sum for %s", name)
+		} else {
+			// Get latest properties from container to get the md5 of blob
+			prop, err := blobURL.GetProperties(context.Background(), bb.blobAccCond, bb.blobCPKOpt)
+			if err != nil {
+				log.Warn("BlockBlob::ReadToFile : Failed to get properties of blob %s (%s)", name, err.Error())
+			} else {
+				blobMD5 := prop.ContentMD5()
+				if blobMD5 == nil {
+					log.Warn("BlockBlob::ReadToFile : Failed to get MD5 Sum for blob %s", name)
+				} else {
+					// compare md5 and fail is not match
+					if !reflect.DeepEqual(fileMD5, prop.ContentMD5()) {
```

Comment:
Reuse the variable blobMD5 created above
```suggestion
					if !reflect.DeepEqual(fileMD5, blobMD5) {
```